c10/core/impl/CUDATraceTLS.cpp

```diff
@@ -0,0 +1,37 @@
+#include <c10/core/impl/CUDATraceTLS.h>
+
+#include <mutex>
+
+namespace c10 {
+namespace impl {
+
+static std::atomic<const PyInterpreter*> cudaTraceState;
+
+void CUDATraceTLS::set_trace(const PyInterpreter* trace) {
+  static std::once_flag flag;
+  std::call_once(
+    flag,
+    [&](){ cudaTraceState.store(trace); }
+  );
+  cudaTraceState.store(trace);
```

Comment:
Why is this being called again outside of `call_once`? Probably an edit mishap?

Reply:
Yes indeed, sorry, I don't know how I've missed it.